libmscore/shadownote.cpp

```diff
@@ -151,9 +160,10 @@ void ShadowNote::draw(QPainter* painter) const
             QPointF pos;
             pos.rx() = up == 1 ? (noteheadWidth - (lw / 2)) : lw / 2;
             qreal yOffset = up == 1 ? symStemUpSE(_notehead).y() * magS() : symStemDownNW(_notehead).y() * magS();
-            if(flag != SymId::lastSym) {
-                  pos.ry() -= up * (symHeight(flag) + (posDot.y() != 0 ? posDot.y() + spatium() : 0) + 0.5*spatium());
-                  painter->drawLine(QLineF(pos.x(), yOffset, pos.x(), pos.y() - up * (yOffset + lw/2)));
+            if (flag != SymId::lastSym) {
+                  qreal flagHeight = (_duration.type() < TDuration::DurationType::V_256TH) ? symHeight(flag) : symHeight(flag) / 2;
```

Comment:
Those shadow notes (seen during note input) had way too long stems, for those added durations.
Those stems seem to depend on the Musical font used, with Bravura's even longer durations than those added here have quite long stems, with Emmentaler just these added ones.
With this change it looks better, but this seems to be quick a hack?